src/components/AddStudent.js

```diff
@@ -0,0 +1,122 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import Cookies from 'js-cookie';
+import { SERVER_URL } from '../constants.js'
+import Grid from '@mui/material/Grid';
+
+class AddStudent extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            open: true,
+            //student: {
+            name: '',
+            email: ''
+            //}
+        }
+        //Start open for testing
+    };
+
+    handleClickOpen = () => {
+        this.setState({ open: true });
+    };
+
+    handleClose = () => {
+        this.setState({ open: false });
+    };
+
+    handleChangeName = (event) => {
+        this.setState({
+            name: event.target.value
+        });
+    }
+
+    handleChangeEmail = (event) => {
+        this.setState({
+            email: event.target.value
+        });
+    }
+
+    // Save student and close modal form
+    handleAdd = () => {
+        console.log("state" + JSON.stringify(this.state));
+        this.props.addStudent(this.state);
```

Comment:
do you need to go through your props to call an arrow function that you already have in your component?